scene/resources/skeleton_modification_2d_twoboneik.cpp

```diff
@@ -172,13 +177,22 @@ void SkeletonModification2DTwoBoneIK::_execute(float p_delta) {
 
 		if (isnan(angle_0) || isnan(angle_1)) {
 			// We cannot solve for this angle! Do nothing to avoid setting the rotation (and scale) to NaN.
-		} else {
+		} else if (same_scale_sign) {
 			joint_one_bone->set_global_rotation(angle_atan - angle_0 - joint_one_bone->get_bone_angle());
 			joint_two_bone->set_rotation(-Math_PI - angle_1 - joint_two_bone->get_bone_angle() + joint_one_bone->get_bone_angle());
+		} else {
+			joint_one_bone->set_global_rotation(angle_atan + angle_0 + joint_one_bone->get_bone_angle());
+			joint_two_bone->set_rotation(-Math_PI - angle_1 - joint_two_bone->get_bone_angle() + joint_one_bone->get_bone_angle());
 		}
+
 	} else {
-		joint_one_bone->set_global_rotation(angle_atan - joint_one_bone->get_bone_angle());
-		joint_two_bone->set_global_rotation(angle_atan - joint_two_bone->get_bone_angle());
+		if (same_scale_sign) {
+			joint_one_bone->set_global_rotation(angle_atan - joint_one_bone->get_bone_angle());
+			joint_two_bone->set_global_rotation(angle_atan - joint_two_bone->get_bone_angle());
+		} else {
+			joint_one_bone->set_global_rotation(angle_atan + joint_one_bone->get_bone_angle());
+			joint_two_bone->set_global_rotation(angle_atan + joint_two_bone->get_bone_angle());
+		}
```

Comment:
Same remark here except you already got a big else block so you'd just have to use the sign multiplier to factorize.

Unless there's a particular policy on this file that developers prefer explicit full line operations with +/- rather than -1/+1 multipliers to make it easier to read? I haven't checked other math code so I don't know what style they picked.

Reply:
I'm holding back with this because could impact the readability and I really want to maintain clear which is the main logic and which is the exception (different scales).  

I don't think that would be easy to understand from something like this:  
```
angle_atan + joint_one_bone->get_bone_angle() * rotation_dir
```

Or this:  
```
angle_atan + (angle_0 + joint_one_bone->get_bone_angle()) * rotation_dir
```

Reply:
Makes sense, you can keep the else case then. In this case, for consistency you can also keep the else case above and don't use a multiplier. You can add a comment indicating which one is the more rare case if you want.